### PR TITLE
disable coupon with dynamic price

### PR DESCRIPTION
**Pull request by poojarazor**

### CODE REVIEW

**Review by angbpy — APPROVED** [review]

File: includes/api/order.php [hunk 1]
```diff
@@ -119,8 +119,31 @@ function createWcOrder(WP_REST_Request $request)
...
+        // Woo dynamic discount price plugin
+        if(is_plugin_active('yith-woocommerce-dynamic-pricing-and-discounts-premium/init.php')) {
+
+            foreach ($order->get_items() as $itemId => $item) {
+                $dynamicRules = $item->get_meta('_ywdpd_discounts');
+
+                foreach ($dynamicRules['applied_discounts'] as $appliedDiscount) {
+                    if (isset( $appliedDiscount['set_id'])){
+                        $ruleId = $appliedDiscount['set_id'];
+                        $rule    = ywdpd_get_rule($ruleId);
```
Comment (ChetanGN) [inline includes/api/order.php:136]: @poojarazor Do we need to initiate a class? to invoke this function?

File: includes/api/order.php [hunk 2]
```diff
@@ -216,6 +239,10 @@ function createWcOrder(WP_REST_Request $request)
         $response['redirect']            = true;
         $response['one_click_checkout']  = true;
 
+        if($disableCouponFlag == true){
+           $response['show_coupons']  = false;
```
Comment (ChetanGN) [inline includes/api/order.php:243]: Do we need to pass default value?

File: includes/api/order.php [hunk 3]
```diff
@@ -119,8 +119,31 @@ function createWcOrder(WP_REST_Request $request)
...
-    if ($order) {
+    
+    if($order){
+
+        $disableCouponFlag = false;
+
+        // Woo dynamic discount price plugin
+        if(is_plugin_active('yith-woocommerce-dynamic-pricing-and-discounts-premium/init.php')) {
+
+            foreach ($order->get_items() as $itemId => $item) {
```
Comment (ChetanGN) [inline includes/api/order.php:130]: Can we get the TCD's for this PR?

File: includes/api/order.php (hunk 1 above)
Comment (poojarazor) [reply includes/api/order.php:136]: it's a public function , no need to initiate any class

File: includes/api/order.php (hunk 2 above)
Comment (poojarazor) [reply includes/api/order.php:243]: no required

File: includes/api/order.php (hunk 3 above)
Comment (poojarazor) [reply includes/api/order.php:130]: Discussed with Gaurav he said  already test case is written for this plugin so no separate TCD required , only I have to do feature testing from my side.

**Pull request merged**